### PR TITLE
Change form

### PR DESCRIPTION
**Pull request by JohnnP**
### Description

Field are always controlled. Like that, when we have a specific onChange for a field. We haven't to set an onChange for all field

### CODE REVIEW

File: src/common/field/index.js [hunk 1]
```diff
@@ -42,7 +42,15 @@ const FieldMixin = {
...
-        this.setState({value: newProps.value, error: newProps.error, values: newProps.values});
+        const {value, values} = this.state;
+        const newState = { value: newProps.value };
+        if (value !== newProps.value || values !== newProps.values) {
+            newState.error = null;
+        }
+        if (newProps.error) {
+            newState.error = newProps.error;
+        }
+        this.setState(newState);
```
Comment (Ephrame) [inline src/common/field/index.js]: @JohnnP You never set values ?

Comment (JohnnP) [inline src/common/field/index.js]: Yeah, you right it's could be problematic. I gonna change that

**Comment by Hartorn** [comment]
This should fix #1347.

**Comment by JohnnP** [comment]
This PR need to be completed. Indeed when we cancel an edition. the field was undefined before it, will keep the value set during edition instead of become undefined.

Probrably use the definitionPath to get the entity and set all value undefined at null when getStateFromStore fix this issue.

**Pull request closed without merging**